pkg/cloud/aws/services/userdata/controlplane.go

```diff
@@ -0,0 +1,83 @@
+// Copyright © 2018 The Kubernetes Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package userdata
+
+import (
+	"bytes"
+	"text/template"
+)
+
+const (
+	controlPlaneBashScript = `{{.Header}}
+
+mkdir -p /etc/kubernetes/pki
+
+echo '{{.CACert}}' > /etc/kubernetes/pki/ca.crt
+echo '{{.CAKey}}' > /etc/kubernetes/pki/ca.key
+
+PRIVATE_IP=$(curl http://169.254.169.254/latest/meta-data/local-ipv4)
+
+cat >/tmp/kubeadm.yaml <<EOF
+---
+apiVersion: kubeadm.k8s.io/v1alpha3
+kind: ClusterConfiguration
+apiServerCertSANs:
+  - "$PRIVATE_IP"
+  - "{{.ELBAddress}}"
+controlPlaneEndpoint: "{{.ELBAddress}}:6443"
+clusterName: "{{.ClusterName}}"
+networking:
+  dnsDomain: "{{.ServiceDomain}}"
+  podSubnet: "{{.PodSubnet}}"
+  serviceSubnet: "{{.ServiceSubnet}}"
+kubernetesVersion: "{{.KubernetesVersion}}"
+---
+apiVersion: kubeadm.k8s.io/v1alpha3
+kind: InitConfiguration
+nodeRegistration:
+  criSocket: /var/run/containerd/containerd.sock
+EOF
+
+kubeadm init --config /tmp/kubeadm.yaml
+`
+)
+
+type ControlPlaneInput struct {
+	userdata
+
+	CACert            string
+	CAKey             string
+	ELBAddress        string
+	ClusterName       string
+	PodSubnet         string
+	ServiceDomain     string
+	ServiceSubnet     string
+	KubernetesVersion string
+}
+
+func NewControlPlane(input *ControlPlaneInput) (string, error) {
```

Comment:
This looks pretty much the same as NewBastion except for the template that is being parsed...would probably be fine to have a function like `func NewNodeScript(kind string, data interface{}) (string, error)`